### PR TITLE
ESIL: add floating point instructions

### PR DESCRIPTION
**Pull request by zamaudio**
Adds mul, div, add, sub and eq variants.

Signed-off-by: Damien Zammit <damien@zamaudio.com>

### CODE REVIEW

File: libr/anal/esil.c [hunk 1]
```diff
@@ -14,6 +14,11 @@
 		R_BIT_UNSET (&esil->flags, FLG (x));\
 	}
 
+typedef union {
```
Comment (radare) [inline libr/anal/esil.c]: its ok to use unions in internal apis, not public ones, otherwise interfacing with other programming languages is a pain

Comment (zamaudio) [inline libr/anal/esil.c]: See r2r # 382 for tests

**Comment by zamaudio** [comment]
Should there exist separate esil instructions for floats vs integers?  such as +F and /F and *F etc?  I don't know but the way I have implemented it for constants means that you lose the type of the value after the first instruction, but don't need separate handling for float...

**Comment by radare** [comment]
No, i think we should reuse current ones

**Comment by condret** [comment]
sorry, i'm late.
1. I think we need instructions for floating point, because esil is not typesafe. Currently everything in esil internal is ut64 implicit. This either should become explicit or we need special operations for floating-point or esil will become as shitty as javascript.
2. How does floating-point affect the internal vars, that provide the information required for setting flag-registers?
3. Explicit typing (we allready do that somehow for memory-access) should make automated analysis and decompilation slightly easier, because we know the what kind of data we handle (int vs float). Having special operations for floating-point provides the same benefit

**Comment by zamaudio** [comment]
@condret can you take a look at #4799 i had a different idea that you may be interested in

**Pull request merged**